grails-app/controllers/org/pih/warehouse/product/ProductController.groovy

```diff
@@ -182,7 +182,8 @@ class ProductController {
             }
         }
 
-        productInstance.validateRequiredFields()
+        List<ProductField> ignoreFields = location.isAccountingRequired() ? [] : [ProductField.GL_ACCOUNT]
+        productInstance.validateRequiredFields(ignoreFields)
```

Comment:
You could also do here a location-based validation: `validateRequiredFieldsInLocation(location)`
```groovy
    Boolean validateRequiredFieldsInLocation(Location location) {
        if (location.isAccountingRequired()) {
            return validateRequiredFields([])
        }

        return validateRequiredFields([ProductField.GL_ACCOUNT])
    }
```
Which could be potentially expanded by other supported activities if needed.

Reply:
looks good, I'll do that